torch/csrc/jit/tensorexpr/bounds_overlap.h

```diff
@@ -61,6 +63,12 @@ enum OverlapKind { ContainedOrEqual, Contains, PartialOverlap, NoOverlap };
 // dimension.
 OverlapKind TORCH_API boundOverlap(Bound A, Bound B);
 
+// Return true if the logic as follows is always satisfied:
+//    (a < b) / (a <= b) / (a > b) / (a >= b) / (a == b) / (a != b)
+// Otherwise, it returns false.
+bool TORCH_API
+compareBound(const Bound& a, const Bound& b, BoundCompareResult& cmp_result);
```

Comment:
Again, we do not use `BoundCompareResult&` to return results from a function. It is something that can be easily overlooked while reading the code. Can you replace this with a pointer instead?